optimum/intel/neural_compressor/modeling_base.py

```diff
@@ -0,0 +1,512 @@
+            model_save_dir = None
+            model_kwargs = {
+                "revision": revision,
+                "use_auth_token": use_auth_token,
+                "cache_dir": cache_dir,
+                "local_files_only": local_files_only,
+                "force_download": force_download,
+                "torch_dtype": torch_dtype,
+            }
+            task = cls.export_feature
+            if config.torch_dtype != "int8":
+                model = TasksManager.get_model_from_task(task, model_id, **model_kwargs)
+            else:
+                state_dict_path = kwargs.get("state_dict_path", None)
+                import copy
+
+                from transformers.models.auto.auto_factory import _get_model_class
+                from transformers.utils import TRANSFORMERS_CACHE, is_offline_mode
+
+                model_class = _get_model_class(config, cls.auto_model_class._model_mapping)
+                keys_to_ignore_on_load_unexpected = copy.deepcopy(
+                    getattr(model_class, "_keys_to_ignore_on_load_unexpected", None)
+                )
+                keys_to_ignore_on_load_missing = copy.deepcopy(
+                    getattr(model_class, "_keys_to_ignore_on_load_missing", None)
+                )
+                # Avoid unnecessary warnings resulting from quantized model initialization
+                quantized_keys_to_ignore_on_load = [
+                    r"zero_point",
+                    r"scale",
+                    r"packed_params",
+                    r"constant",
+                    r"module",
+                    r"best_configure",
+                    r"max_val",
+                    r"min_val",
+                    r"eps",
+                    r"fake_quant_enabled",
+                    r"observer_enabled",
+                ]
+                if keys_to_ignore_on_load_unexpected is None:
+                    model_class._keys_to_ignore_on_load_unexpected = quantized_keys_to_ignore_on_load
+                else:
+                    model_class._keys_to_ignore_on_load_unexpected.extend(quantized_keys_to_ignore_on_load)
+                missing_keys_to_ignore_on_load = [r"weight", r"bias"]
+                if keys_to_ignore_on_load_missing is None:
+                    model_class._keys_to_ignore_on_load_missing = missing_keys_to_ignore_on_load
+                else:
+                    model_class._keys_to_ignore_on_load_missing.extend(missing_keys_to_ignore_on_load)
+
+                try:
+                    model = model_class.from_pretrained(model_id, **kwargs)
+                except AttributeError:
+                    init_contexts = [no_init_weights(_enable=True)]
+                    with ContextManagers(init_contexts):
+                        model = model_class(config, **kwargs)
+
+                model_class._keys_to_ignore_on_load_unexpected = keys_to_ignore_on_load_unexpected
+                model_class._keys_to_ignore_on_load_missing = keys_to_ignore_on_load_missing
```

Comment:
This already exists (integrated in [`INCModel`](https://github.com/huggingface/optimum-intel/blob/v1.8.1/optimum/intel/neural_compressor/quantization.py#L500)) please refactorize to avoid any redundancy

Reply:
In the future, we want to remove INCModel from quantization.py, we want that all model class will inherit INCBaseModel.